Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,469 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using QuantConnect.Configuration;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using TradingApi.Bitfinex;
+using TradingApi.ModelObjects.Bitfinex.Json;
+
+namespace QuantConnect.Brokerages.Bitfinex
+{
+
+    /// <summary>
+    /// Bitfinex exchange REST integration.
+    /// </summary>
+    public partial class BitfinexBrokerage : Brokerage, IDataQueueHandler
+    {
+
+        #region Declarations
+        /// <summary>
+        /// Ticks collection
+        /// </summary>
+        protected List<Tick> ticks = new List<Tick>();
+        CancellationTokenSource _tickerToken;
+        /// <summary>
+        /// Divisor for prices. Scales prices/volumes to allow trades on 0.01 of unit
+        /// </summary>
+        protected decimal divisor = 100;
+        readonly object _fillLock = new object();
+        const string buy = "buy";
+        const string sell = "sell";
+        /// <summary>
+        /// Currently limited to BTCUSD
+        /// </summary>
+        protected Symbol symbol = Symbol.Create("BTCUSD", SecurityType.Forex, Market.Bitfinex);
+        /// <summary>
+        /// List of known orders
+        /// </summary>
+        public ConcurrentDictionary<int, Order> CachedOrderIDs = new ConcurrentDictionary<int, Order>();
+        /// <summary>
+        /// List of filled orders
+        /// </summary>
+        protected readonly FixedSizeHashQueue<int> filledOrderIDs = new FixedSizeHashQueue<int>(10000);
+        /// <summary>
+        /// List of unknown orders
+        /// </summary>
+        protected readonly FixedSizeHashQueue<int> unknownOrderIDs = new FixedSizeHashQueue<int>(1000);
+        /// <summary>
+        /// Name of wallet
+        /// </summary>
+        protected string wallet;
+        const string _exchange = "bitfinex";
+        /// <summary>
+        /// Api Key
+        /// </summary>
+        protected string apiKey;
+        /// <summary>
+        /// Api Secret
+        /// </summary>
+        protected string apiSecret;
+        #endregion
+
+        public TradingApi.Bitfinex.BitfinexApi RestClient { get; set; }
+
+
+        /// <summary>
+        /// Create bitfinex brokerage
+        /// </summary>
+        public BitfinexBrokerage()
+            : base("bitfinex")
+        {
+            this.Initialize();
+        }
+
+        private void Initialize()
+        {
+
+            //todo: also stored in BrokerageData
+            apiSecret = Config.Get("bitfinex-api-secret");
+            apiKey = Config.Get("bitfinex-api-key");
```

Comment:
These initialization parameters should be accepted as constructor arguments. We could even include the `RestClient` as a constructor argument. This will provide consumers of this class more flexibility and remove the need for public get/set